### DISCUSSION TITLE
Code Quality Testing

### DISCUSSION THREAD

**Discussion by st3phani** [markshust/docker-magento#493] [Ideas]
I think it is necessary to test the code quality for all types of files: 

- Php
- Phtml
- Xml
- JS
- Json
- Html
- CSS
- Less

All such tools that can analyze the quality of the code (phpcs, phpmd, ...).

You can make a **CLI Command** and a **Watch Command**.

I mean to be tested on everything that is custom created, possibly to be a configuration in which to mention the routes from `vendor/package` and `app/code/Package`.

**Comment by wizziLalev** [comment]
I am with the same request - usage of phpcs, php-cs-fixer, etc. is something that miss a lot. Any ideas how to make them available based on currently documented setup?

**Comment by markshust** [comment]
I deleted my previous posts so that all the info can live in one spot (where it will continue to be updated & maintained). I published a new lesson on how to set up Code Quality Tools in PhpStorm at https://courses.m.academy/courses/set-up-magento-2-development-environment-docker/lectures/36974633

**Reply by yurylapi** [reply]
> I deleted my previous posts so that all the info can live in one spot (where it will continue to be updated & maintained). I published a new lesson on how to set up Code Quality Tools in PhpStorm at https://courses.m.academy/courses/set-up-magento-2-development-environment-docker/lectures/36974633

Hi there, I have done everything according to your documentation and  got the error

`
![image](https://user-images.githubusercontent.com/44709956/148745653-125e9e29-7ba9-4e45-8bbb-b3f462e35893.png)
`

Do you know why there is some weird container is trying to be created and even after it is created and run phpcs and other tools are failed?

**Reply by wizziLalev** [reply]
I've got the same - just try to restart everything, that fixed it - maybe PHPStorm cannot make it from 1st run...
